sys/include/net/netopt.h

```diff
@@ -248,6 +248,16 @@ typedef enum {
      * @note    Interfaces are not meant to respond to that.
      */
     NETOPT_NUMOF,
+
+    /**
+     * @brief   add/remove an address to/from the link layer filter list
+     *
+     * Get expexts a pointer to a @ref l2filter_t struct. Set expexts a pointer
+     * to a hardware address, usually a uint8_t array. These options are used for both,
+     * blacklisting and whitelisting.
+     */
+    NETOPT_L2FILTER,
+    NETOPT_L2FILTER_RM, /**< get will always return -ENOTSUP */
```

Comment:
Btw, I think you need to extend [`_netopt_strmap`](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/crosslayer/netopt/netopt.c#L26-L62), too.